contents/android/IntroToAndroid.md

```diff
@@ -0,0 +1,145 @@
+
+To read more about the MVVM architecture in Android apps, see this [tutorial](https://proandroiddev.com/mvvm-architecture-viewmodel-and-livedata-part-1-604f50cda1). For a comparison of the three patterns, see [this article at Realm's Academy](https://academy.realm.io/posts/eric-maxwell-mvc-mvp-and-mvvm-on-android/).
+
+### 2.4 Resources
+
+Apps rely a lot on external resources and Android provides an easy way of accessing these resources.
+
+Android manages external resources in one central place. Any external resources that the app requires such as images, XML layout files, fonts, values such as strings and colors should be placed in the `res/` directory of the Android project. The resource then can be accessed by referencing its resource ID. Its resource ID is composed of the [resource type](https://developer.android.com/guide/topics/resources/available-resources.html) and the resource name. `R.drawable.myimage` is an example of a way to access an image resource in code. `@string/submit` is an example of a way to access a string value from XML. Read more about accessing resources [here](https://developer.android.com/guide/topics/resources/accessing-resources.html).
+
+### 2.5 Testing
+
+Testing used to be a nightmare for app development but since 2017, Test-driven development was [encouraged](https://www.youtube.com/watch?v=pK7W5npkhho) by Google in their Google I/O 2017 due to the release of the `Android Testing Support Library`. You can read more about the library [here](https://developer.android.com/topic/libraries/testing-support-library/index.html). Now that we can do testing easily, we can build more reliable apps!
+
+#### Unit Testing and Mocking
+Mocking is usually needed in unit testing to simulate the behaviour of real objects (that are dependencies) to verify the behaviour of the object that you are testing. In Android app development, [Mockito](http://site.mockito.org/), a popular mocking library for unit tests in Java, is often used. However, due to the difficulty of mocking the Android SDK to create unit tests, [Robolectric](http://robolectric.org/) was developed to solve this issue. Robolectric can still be used alongside Mockito. The benefit of Robolectric is that it handles emulation UI code such that tests that rely on the UI can be run on the Java Virtual Machine (JVM) rather than having to run them on an emulator. In other words, tests can be written and run faster.
+
+#### Integration Testing
+Integration testing and end-to-end testing is usually done using [Espresso](https://developer.android.com/training/testing/espresso/index.html). You can even record your own UI tests using the [Espresso Test Recorder](https://developer.android.com/studio/test/espresso-test-recorder.html) to save time! Be sure to check out the Espresso [cheatsheet](https://developer.android.com/training/testing/espresso/cheat-sheet.html) for easy writing of tests.
+
+## 4. Why should I make native Android apps?
```

Comment:
Move this section up? Suggested outline:
1. What is Android?
1. Why Android? (this section can also mention negative points too, to give a balanced view)
1. Getting started
1. Basic Android Concepts (I'm still of the opinion that there is no need for this section, but if you must include it, mention that it is a 'a quick preview of some concepts to whet the appetite of the reader', and make it shorter than the current version)
1. Useful Android Developer Tools (you can use this section to mention things such as testing tools to show that Android devs are well-supported by tools)
1. Further reading